run_dir/static/js/running_notes_component.js

```diff
@@ -0,0 +1,442 @@
+                new_note_form.classList.add('show');
+                new_note_caret.classList.remove('fa-caret-right');
+                new_note_caret.classList.add('fa-caret-down');
+                // focus on the textarea
+                this.$refs.form_note_text_field.focus();
+            }
+        },
+        setCurrentPosition() {
+            let textarea = this.$refs.form_note_text_field;
+            if (textarea === undefined) {
+                return {}
+            }
+            this.dropdown_position = this.$root.getDropdownPositionHelper(textarea, 100);
+        },
+        setFilter(filter) {
+            this.category_filter = filter
+        },
+        submitRunningNote() {
+            this.submitting = true;
+            if (this.form_note_text === '') {
+                alert("Error: No running note entered.");
+                return
+            }
+            if (this.form_categories.length === 0) {
+                if (!confirm("Are you sure that you want to submit without choosing a category?")) {
+                    return
+                }
+            }
+
+            if(["flowcell", "workset", "flowcell_ont"].includes(this.note_type)){
+                if((this.note_type==="flowcell" || this.note_type==="flowcell_ont") && !this.form_categories.includes("Flowcell")){
+                  this.form_categories.push("Flowcell")
+                }
+                else if(this.note_type==="workset" && !this.form_categories.includes("Workset")){
+                  this.form_categories.push("Workset")
+                }
+            }
+
+            let post_body = {
+                note: this.form_note_text,
+                categories: this.form_categories,
+                note_type: this.note_type
+            }
+
+
+            axios
+                .post('/api/v1/running_notes/' + this.partition_id, post_body)
+                .then(response => {
+                    this.fetchAllRunningNotes(this.partition_id)
+                    this.form_note_text = ''
+                    this.form_categories = []
+                    this.submitting = false
+                })
+                .catch(error => {
+                    alert('Unable to submit running note, please try again or contact a system administrator.')
+                    this.submitting = false
+                })
+        },
+        suggestTaggedUsers(event) {
+            this.setCurrentPosition();
+            let form_note_text_field = this.$refs.form_note_text_field;
+            let cursor_position = form_note_text_field.selectionStart;
+            let text_before_cursor = form_note_text_field.value.substring(0, cursor_position);
+            let current_word = text_before_cursor.split(' ').pop();
+
+            if (current_word.startsWith('@')) {
+                current_word = current_word.substring(1).toLowerCase();
+                let user_suggestions = this.all_users.filter(user => user.startsWith(current_word));
```

Comment:
This could return matches of surname too
```suggestion
                let user_suggestions = this.all_users.filter(user => user.includes(current_word));
```